# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1316651039 14400
#      Wed Sep 21 20:23:59 2011 -0400
# Node ID e0a3cca6e6770e44be50bb24fffc9b19da4b88b4
# Parent  8a124aeb7ded88988235f7f62e411ff187fed88d
Add tolerances to binopdf and tpdf tests.

* statistics/distributions/binopdf.m: Add tolerance to accommodate MacOS's
gammal.

* statistics/distributions/tpdf.m: Add tolerance.

diff --git a/scripts/statistics/distributions/binopdf.m b/scripts/statistics/distributions/binopdf.m
--- a/scripts/statistics/distributions/binopdf.m
+++ b/scripts/statistics/distributions/binopdf.m
@@ -61,25 +61,30 @@ function pdf = binopdf (x, n, p)
   else
     pdf(k) = exp (gammaln (n(k)+1) - gammaln (x(k)+1) - gammaln (n(k)-x(k)+1)
                   + x(k).*log (p(k)) + (n(k)-x(k)).*log (1-p(k)));
   endif
 
 endfunction
 
 
-%!shared x,y
+%!shared x,y,tol
+%! if (ismac ())
+%!   tol = eps ();
+%! else
+%!   tol = 0;
+%! endif
 %! x = [-1 0 1 2 3];
 %! y = [0 1/4 1/2 1/4 0];
-%!assert(binopdf (x, 2*ones(1,5), 0.5*ones(1,5)), y);
-%!assert(binopdf (x, 2, 0.5*ones(1,5)), y);
-%!assert(binopdf (x, 2*ones(1,5), 0.5), y);
+%!assert(binopdf (x, 2*ones(1,5), 0.5*ones(1,5)), y, tol);
+%!assert(binopdf (x, 2, 0.5*ones(1,5)), y, tol);
+%!assert(binopdf (x, 2*ones(1,5), 0.5), y, tol);
 %!assert(binopdf (x, 2*[0 -1 NaN 1.1 1], 0.5), [0 NaN NaN NaN 0]);
 %!assert(binopdf (x, 2, 0.5*[0 -1 NaN 3 1]), [0 NaN NaN NaN 0]);
-%!assert(binopdf ([x, NaN], 2, 0.5), [y, NaN]);
+%!assert(binopdf ([x, NaN], 2, 0.5), [y, NaN], tol);
 
 %% Test class of input preserved
 %!assert(binopdf (single([x, NaN]), 2, 0.5), single([y, NaN]));
 %!assert(binopdf ([x, NaN], single(2), 0.5), single([y, NaN]));
 %!assert(binopdf ([x, NaN], 2, single(0.5)), single([y, NaN]));
 
 %% Test input validation
 %!error binopdf ()
